src/org/traccar/protocol/ArnaviBinaryProtocolDecoder.java

```diff
@@ -0,0 +1,273 @@
+    private static final byte PACKET_TEXT = 0x03;
+    private static final byte PACKET_FILE = 0x04;
+    private static final byte PACKET_DATA_BINARY = 0x06;
+
+    private static final byte TAG_VOLTAGE_SUPPLY = 1;
+    private static final byte TAG_ID_KEY = 2;
+    private static final byte TAG_LATITUDE = 3;
+    private static final byte TAG_LONGITUDE = 4;
+    private static final byte TAG_COORD_PARAMS = 5;
+    private static final byte TAG_DEVICE_STATUS = 9;
+    private static final byte TAG_FREQ_0 = 20;
+    private static final byte TAG_FREQ_1 = 21;
+    private static final byte TAG_FREQ_2 = 22;
+    private static final byte TAG_FREQ_3 = 23;
+    private static final byte TAG_FREQ_4 = 24;
+    private static final byte TAG_FREQ_5 = 25;
+    private static final byte TAG_FREQ_6 = 26;
+    private static final byte TAG_FREQ_7 = 27;
+    private static final byte TAG_VOLT_0 = 30;
+    private static final byte TAG_VOLT_1 = 31;
+    private static final byte TAG_VOLT_2 = 32;
+    private static final byte TAG_VOLT_3 = 33;
+    private static final byte TAG_VOLT_4 = 34;
+    private static final byte TAG_VOLT_5 = 35;
+    private static final byte TAG_VOLT_6 = 36;
+    private static final byte TAG_VOLT_7 = 37;
+    private static final byte TAG_LLS_0 = 70;
+    private static final byte TAG_LLS_1 = 71;
+    private static final byte TAG_LLS_2 = 72;
+    private static final byte TAG_LLS_3 = 73;
+    private static final byte TAG_LLS_4 = 74;
+    private static final byte TAG_LLS_5 = 75;
+    private static final byte TAG_LLS_6 = 76;
+    private static final byte TAG_LLS_7 = 77;
+    private static final byte TAG_LLS_8 = 78;
+    private static final byte TAG_LLS_9 = 79;
+
+
+    private List<Position> positions;
+
+
+    private static int crc8(byte[] data) {
+        byte crc = 0;
+        for (byte b : data) {
+            crc += b;
+        }
+        return (crc & 0xff);
+    }
+
+    private void sendConfirmationHeader(Channel channel, int headerVersion) {
+        if (headerVersion != PROTOCOL_VERSION1 && headerVersion != PROTOCOL_VERSION2) {
+            return;
+        }
+        ChannelBuffer reply;
+        if (headerVersion == PROTOCOL_VERSION1) {
+            reply = ChannelBuffers.directBuffer(ByteOrder.LITTLE_ENDIAN, 4);
+            reply.writeInt(0x7B00007D);
+        } else {
+            byte[] timestampBytes = ByteBuffer.allocate(4).putInt((int) Instant.now().getEpochSecond()).array();
+            reply = ChannelBuffers.directBuffer(ByteOrder.LITTLE_ENDIAN, 9);
+            reply.writeByte((byte) 0x7B);
+            reply.writeByte((byte) 0x04);
+            reply.writeByte((byte) 0x00);
+            reply.writeByte((byte) crc8(timestampBytes));
+            reply.writeBytes(timestampBytes);
+            reply.writeByte((byte) 0x7D);
+        }
+        if (channel != null) {
+            channel.write(reply);
+        }
+    }
+
+    private void sendConfirmationPackage(Channel channel, int packageCounter) {
```

Comment:
What's the difference between confirmation header and package? Names are confusing.